### ISSUE TITLE
Optional doesn't work with union None from 3.10

### ISSUE THREAD

**Issue by pauleveritt** [Finistere/antidote#38]
Python 3.10 allows writing this:

```python
@inject
def optional_greeter(greeter: MissingGreeter | None = inject.me()) -> str:
```

...instead of:

```python
@inject
def optional_greeter(greeter: Optional[MissingGreeter] = inject.me()) -> str:
```

However, this results in:

```
../src/antidote/core/_injection.py:110: in _build_injection_blueprint
    annotated = _build_from_annotations(arguments)
../src/antidote/core/_injection.py:138: in _build_from_annotations
    dependency = extract_annotated_arg_dependency(arg)
../src/antidote/core/_annotations.py:97: in extract_annotated_arg_dependency
    raise TypeError(f"Cannot use marker @inject.me with non class type hint: examples.injection_flavors.MissingGreeter | None
```

**Comment by Finistere** [comment]
Indeed, thanks for reporting. Releasing a fix.

**Comment by Finistere** [comment]
Fixed with the release `1.2.0`.
